File: 1046-Last Stone Weight/solution.py
class Solution(object):
   def lastStoneWeight(self, stones):
      """
      :type stones: List[int]
      :rtype: int
      """
      if len(stones) ==0:
         return 0
      if len(stones) == 1:
         return 1
      while len(stones)>1:
         stones.sort()
         s1,s2=stones[-1],stones[-2]
         if s1==s2:
            stones.pop(-1)
            stones.pop(-1)
         else:
            s1 = abs(s1-s2)
            stones.pop(-1)
            stones[-1] = s1
      if len(stones):
         return stones[-1]
      return 0

File: 1046-Last Stone Weight/solution.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int>pq;
      //Sort the array using heap/Priority Queue
        for(int i=0;i<stones.size();i++){
            pq.push(stones[i]);
        }
      //If only one element return that element
        if (pq.size()==1) return pq.top();
      //Repeat the process till we left wtho zero or one element.
        while(pq.size()>1){
          //Pick two greater elements
         int   x=pq.top();
            pq.pop();
         int   y=pq.top();
            pq.pop();
            if(x==y) continue;  
            else pq.push(x-y);
            
        }
        int ans;
        if (!pq.empty()) ans=pq.top();
        else ans=0;
        return ans;
    }
};